rewrite-java/src/test/java/org/openrewrite/java/OpenRewriteArchitectureTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import com.tngtech.archunit.core.domain.JavaClasses;
+import com.tngtech.archunit.core.domain.JavaPackage;
+import com.tngtech.archunit.core.importer.ClassFileImporter;
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.lang.*;
+import org.junit.jupiter.api.Test;
+import org.openrewrite.internal.lang.NonNullApi;
+
+import static com.tngtech.archunit.base.DescribedPredicate.doNot;
+import static com.tngtech.archunit.core.domain.properties.HasName.Predicates.name;
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.*;
+
+public class OpenRewriteArchitectureTest {
+    JavaClasses importedClasses = new ClassFileImporter()
+      .withImportOption(new ImportOption.DoNotIncludeTests())
+      .importPackages("org.openrewrite.java");
+
+    @Test
+    void packageInfo() {
+        all(packages("org.openrewrite.java"))
+          .that(doNot(name("org.openrewrite.java.internal.grammar")))
+          .should(containAnAnnotatedPackageInfo())
+          .check(importedClasses);
+    }
+
+    @Test
+    void nullness() {
+        noCodeUnits()
+          .should().beAnnotatedWith("org.jetbrains.annotations.NotNull")
+          .orShould().beAnnotatedWith("org.jetbrains.annotations.Nullable")
+          .check(importedClasses);
+    }
+
+    private static ClassesTransformer<JavaPackage> packages(String basePackage) {
+        return new AbstractClassesTransformer<>("packages") {
+            @Override
+            public Iterable<JavaPackage> doTransform(JavaClasses classes) {
+                return classes.getPackage(basePackage).getSubpackagesInTree();
```

Comment:
Didn't know about this one; neat!

Reply:
Used to be called `getAllSubpackages()` pre 1.0.0.